src/enclave/ServiceProvider/ServiceProvider.cpp

```diff
@@ -147,284 +151,208 @@ void ServiceProvider::export_public_key_code(const std::string &filename) {
+  t.seekg(0, std::ios::end);
+  size_t public_key_size = t.tellg();
+  public_key.reserve(public_key_size + 1);
+  t.seekg(0, std::ios::beg);
 
-    std::vector<std::string> messages;
-    ias_check(req.report(payload, content, messages));
-  } catch (const std::runtime_error &e) {
-    if (require_attestation) {
-      throw;
-    }
+  public_key.assign((std::istreambuf_iterator<char>(t)),
+                    std::istreambuf_iterator<char>());
+  public_key.replace(public_key_size, 1, "\0");
+
+  if (!verify_mrsigner((char*)public_key.c_str(),
+                       public_key.size(),
+                       parsed_report.identity.signer_id,
+                       sizeof(parsed_report.identity.signer_id))) {
+    throw std::runtime_error(std::string("failed: mrsigner not equal!"));
   }
 
-  if (!content.empty()) {
-    json::JSON reportObj = json::JSON::Load(content);
+  std::cout << "Signer verification passed\n" ;
 
-    if (!reportObj.hasKey("version")) {
-      throw std::runtime_error("IAS: Report is missing an API version.");
-    }
+  // TODO missing the hash verification step
 
-    unsigned int rversion = (unsigned int)reportObj["version"].ToInt();
-    if (rversion != ias_api_version) {
-      throw std::runtime_error(
-        std::string("IAS: Report version ")
-        + std::to_string(rversion)
-        + std::string(" does not match API version ")
-        + std::to_string(ias_api_version));
-    }
+  // check the enclave's product id and security version
+  if (parsed_report.identity.product_id[0] != 1) {
+    throw std::runtime_error(std::string("identity.product_id checking failed."));
+  }
 
-    // "Extract the attestation status for the enclave.
-    // Decide whether or not to trust the enclave."
-    if (reportObj["isvEnclaveQuoteStatus"].ToString().compare("OK") == 0) {
-      // Enclave is trusted
-    } else if (reportObj["isvEnclaveQuoteStatus"].ToString().compare("CONFIGURATION_NEEDED") == 0) {
-      throw std::runtime_error(
-        "Enclave not trusted. IAS reports CONFIGURATION_NEEDED. Check the BIOS.");
-    } else if (reportObj["isvEnclaveQuoteStatus"].ToString().compare("GROUP_OUT_OF_DATE") != 0) {
-      throw std::runtime_error(
-        "Enclave not trusted. IAS reports GROUP_OUT_OF_DATE. Update the BIOS.");
-    } else {
-      if (require_attestation) {
-        throw std::runtime_error("Enclave not trusted.");
-      }
-    }
+  if (parsed_report.identity.security_version < 1) {
+    throw std::runtime_error(std::string("identity.security_version checking failed."));
+  }
+  
+  // TODO:
+  // "Examine the enclave identity (MRSIGNER), security version and product ID.
+  // Examine the debug attribute and ensure it is not set (in a production environment).
+  // Decide whether or not to trust the enclave and, if provided, the PSE."
+  
+  // TODO:
+  // "Derive the session keys, SK and MK, that should be used to transmit
+  // future messages between the client and server during the session.
```

Comment:
These quotations reference [Intel's guide](https://software.intel.com/en-us/articles/code-sample-intel-software-guard-extensions-remote-attestation-end-to-end-example). Are they still applicable to OpenEnclave?